Library/Homebrew/build.rb

```diff
@@ -190,7 +190,20 @@ def fixopt(f)
   build   = Build.new(formula, options)
   build.install
 rescue Exception => e # rubocop:disable Lint/RescueException
-  Marshal.dump(e, error_pipe)
+  error_hash = JSON.parse e.to_json
```

Comment:
Why convert this to JSON and then to a hash?

Reply:
So that we can handle adding extra information in the `BuildError` case below -- `Exception` doesn't have a `#to_h` or a `#to_hash` method, so I figured it was slightly cleaner to go from JSON to a hash than to duplicate write a monkeypatched method or `Utils` method that does the same.